### PR TITLE
Skip discovery step by informing oidc endpoints

### LINKED ISSUE #763: Skip Discovery Phase and Define Identity Endpoints
### Issue and Steps to Reproduce

Hi team, 

Great job creating this lib, it's really cool and makes it very easy to implement secure SPAs. 

I have a quite hard requirement on my hand and I am seeking clarifications about the best way to implement it. I have an implementation of OIDC to make but the identity server cannot and will not add our domain to CORS. How can we skip the discovery phase and give the actual endpoints that should be called? This identity server supports authorization code and implicit, but I read here (https://github.com/AxaGuilDEv/react-oidc/issues/751) that implicit is not supported. 

Hence the question, how can I skip discovery and define the Authorization, Token, UserInfo and other endpoints to avoid the CORS issue?

### Versions
v5.4.2

#### Expected
```
const configuration = {
  client_id: 'interactive.public.short',
  redirect_uri: 'http://localhost:4200/authentication/callback',
  silent_redirect_uri: 'http://localhost:4200/authentication/silent-callback', // Optional activate silent-signin that use cookies between OIDC server and client javascript to restore the session
  scope: 'openid profile email api offline_access',
  authority: {
    authorization: 'https://demo.identityserver.io/authorization',
    token: 'https://demo.identityserver.io/token',
    userinfo: 'https://demo.identityserver.io/userinfo',
    introspect: 'https://demo.identityserver.io/introspect',
    jwks: 'https://demo.identityserver.io/jwks',
    issuer: 'https://demo.identityserver.io/issuer',
  }
  service_worker_relative_url:'/OidcServiceWorker.js',
  service_worker_only:true,
};
```
#### Actual
Only authority variable is documented.

### Additional Details

- Installed packages:
    "react": "^17.0.2",
    "react-csv": "^2.2.2",
    "react-dom": "^17.0.2",

### PR DESCRIPTION
**Pull request by slaterx**
Fixes #763 

## Before this PR
image

## After this PR
image

### CODE REVIEW

**Comment by guillaume-chervet** [comment]
I have cloned your repository, We have to inestigate, it should work like that :/

**Comment by guillaume-chervet** [comment]
I do not know why, but it work if you do like that. 

Do you have time to finish the PullRequest?


`
    async initAsync(authority) {
        
        return new AuthorizationServiceConfiguration( {authorization_endpoint: "https://demo.duendesoftware.com/connect/authorize",
        end_session_endpoint: "https://demo.duendesoftware.com/connect/endsession",
        revocation_endpoint: "https://demo.duendesoftware.com/connect/revocation",
        token_endpoint: "https://demo.duendesoftware.com/connect/token",
            userinfo_endpoint: "https://demo.duendesoftware.com/connect/userinfo"});
        
        const oidcServerConfiguration = await AuthorizationServiceConfiguration.fetchFromIssuer(authority, new FetchRequestor());
        return oidcServerConfiguration;
    }
`

File: packages/context/src/oidc/vanilla/oidc.ts [hunk 1]
```diff
@@ -295,8 +297,13 @@ export class Oidc {
         });
         return promise;
     }
-    async initAsync(authority) {
-        const oidcServerConfiguration = await AuthorizationServiceConfiguration.fetchFromIssuer(authority, new FetchRequestor());
+    async initAsync(authority, authority_configuration?) {
+        let oidcServerConfiguration
+        if (authority_configuration != null) {
+            oidcServerConfiguration = authority_configuration;
```
Comment (guillaume-chervet) [inline packages/context/src/oidc/vanilla/oidc.ts:303]: return new AuthorizationServiceConfiguration(authority_configuration);

sould fixe your problem

File: packages/context/src/oidc/vanilla/oidc.ts [hunk 2]
```diff
@@ -59,18 +60,19 @@ export interface StringMap {
...
+export type OidcConfiguration = {
     client_id: string,
     redirect_uri: string,
-     silent_redirect_uri?:string,
-     silent_signin_timeout?:number,
+    silent_redirect_uri?:string,
+    silent_signin_timeout?:number,
     scope: string,
     authority: string,
+    authority_configuration?: AuthorizationServiceConfigurationJson,
```
Comment (guillaume-chervet) [inline packages/context/src/oidc/vanilla/oidc.ts:70]: I would not exporse directly AppAuthJS type. I prefer to create a new one to protect the library from internal library change.

**Comment by guillaume-chervet** [comment]
Very nice PullRequest thank you :)

**Comment by slaterx** [comment]
Superseded by #768

**Pull request closed without merging**